scripts/deploy_jenkins.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/sh -xe
+SCRIPTS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+TEMPLATES_DIR="$( cd $SCRIPTS_DIR/templates && pwd )"
+
+oc new-project $PROJECT_NAME
+
+for AGENT in jenkins-tools
```

Comment:
Where is this `jenkins-tools` coming from?

Reply:
This is just an array with 1 thing in "jenkins-tools", if you had more than 1 agent you could add it here. jenkins-tools is a container we used previously that has basic tools pre installed on it, one of which is skopeo which is all we really rely on for the openshift jobs https://github.com/feedhenry/wendy-jenkins-s2i-configuration/blob/master/agent-jenkins-tools/Dockerfile

The container is already built and available on the feedhenry quay repo, i never bothered changing it, so it's just referenced here in the template https://github.com/integr8ly/ci-cd/blob/13ad5dad09eb569719428d4ddfbc539c67956835/scripts/templates/agent-image-template.yml

When Jenkins starts up, the kuberenetes plugin will check for ImageStreams available that have the correct role label applied https://github.com/integr8ly/ci-cd/blob/13ad5dad09eb569719428d4ddfbc539c67956835/scripts/templates/agent-image-template.yml#L22 
If it finds one, it will add that as k8s pod template in the global config  

![image](https://user-images.githubusercontent.com/327352/60817139-fe953800-a192-11e9-85b1-f42699375620.png)

If you request a node matching those labels it will spin up a new pod in OpenShift and run everything inside of that.

Reply:
If it helps, this is where it was taken from https://github.com/feedhenry/wendy-jenkins-s2i-configuration/blob/master/scripts/deploy_jenkins.sh#L50